INFO:uvicorn.error:Will watch for changes in these directories: ['/Users/Acer/Desktop/GitHub Projects/NLP/VideoToPPT/app']
INFO:uvicorn.error:Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
INFO:uvicorn.error:Started reloader process [5495] using WatchFiles
INFO:uvicorn.error:Started server process [5499]
INFO:uvicorn.error:Waiting for application startup.
INFO:uvicorn.error:Application startup complete.
ERROR:uvicorn.error:Exception in ASGI application
Traceback (most recent call last):
  File "/Users/Acer/Desktop/GitHub Projects/NLP/env/lib/python3.12/site-packages/uvicorn/protocols/http/httptools_impl.py", line 409, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Acer/Desktop/GitHub Projects/NLP/env/lib/python3.12/site-packages/uvicorn/middleware/proxy_headers.py", line 60, in __call__
    return await self.app(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Acer/Desktop/GitHub Projects/NLP/env/lib/python3.12/site-packages/fastapi/applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "/Users/Acer/Desktop/GitHub Projects/NLP/env/lib/python3.12/site-packages/starlette/applications.py", line 112, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/Users/Acer/Desktop/GitHub Projects/NLP/env/lib/python3.12/site-packages/starlette/middleware/errors.py", line 187, in __call__
    raise exc
  File "/Users/Acer/Desktop/GitHub Projects/NLP/env/lib/python3.12/site-packages/starlette/middleware/errors.py", line 165, in __call__
    await self.app(scope, receive, _send)
  File "/Users/Acer/Desktop/GitHub Projects/NLP/env/lib/python3.12/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "/Users/Acer/Desktop/GitHub Projects/NLP/env/lib/python3.12/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "/Users/Acer/Desktop/GitHub Projects/NLP/env/lib/python3.12/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "/Users/Acer/Desktop/GitHub Projects/NLP/env/lib/python3.12/site-packages/starlette/routing.py", line 715, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/Users/Acer/Desktop/GitHub Projects/NLP/env/lib/python3.12/site-packages/starlette/routing.py", line 735, in app
    await route.handle(scope, receive, send)
  File "/Users/Acer/Desktop/GitHub Projects/NLP/env/lib/python3.12/site-packages/starlette/routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "/Users/Acer/Desktop/GitHub Projects/NLP/env/lib/python3.12/site-packages/starlette/routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "/Users/Acer/Desktop/GitHub Projects/NLP/env/lib/python3.12/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "/Users/Acer/Desktop/GitHub Projects/NLP/env/lib/python3.12/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "/Users/Acer/Desktop/GitHub Projects/NLP/env/lib/python3.12/site-packages/starlette/routing.py", line 73, in app
    response = await f(request)
               ^^^^^^^^^^^^^^^^
  File "/Users/Acer/Desktop/GitHub Projects/NLP/env/lib/python3.12/site-packages/fastapi/routing.py", line 301, in app
    raw_response = await run_endpoint_function(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Acer/Desktop/GitHub Projects/NLP/env/lib/python3.12/site-packages/fastapi/routing.py", line 214, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Acer/Desktop/GitHub Projects/NLP/env/lib/python3.12/site-packages/starlette/concurrency.py", line 37, in run_in_threadpool
    return await anyio.to_thread.run_sync(func)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Acer/Desktop/GitHub Projects/NLP/env/lib/python3.12/site-packages/anyio/to_thread.py", line 56, in run_sync
    return await get_async_backend().run_sync_in_worker_thread(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Acer/Desktop/GitHub Projects/NLP/env/lib/python3.12/site-packages/anyio/_backends/_asyncio.py", line 2470, in run_sync_in_worker_thread
    return await future
           ^^^^^^^^^^^^
  File "/Users/Acer/Desktop/GitHub Projects/NLP/env/lib/python3.12/site-packages/anyio/_backends/_asyncio.py", line 967, in run
    result = context.run(func, *args)
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/Acer/Desktop/GitHub Projects/NLP/VideoToPPT/app/main.py", line 32, in get_new_session
    command = NewSessionCommand()
              ^^^^^^^^^^^^^^^^^^^
TypeError: Can't instantiate abstract class NewSessionCommand without an implementation for abstract method 'process'
INFO:watchfiles.main:1 change detected
WARNING:uvicorn.error:WatchFiles detected changes in 'commands/session.py'. Reloading...
INFO:uvicorn.error:Shutting down
INFO:uvicorn.error:Waiting for application shutdown.
INFO:uvicorn.error:Application shutdown complete.
INFO:uvicorn.error:Finished server process [5499]
INFO:watchfiles.main:1 change detected
INFO:uvicorn.error:Started server process [5575]
INFO:uvicorn.error:Waiting for application startup.
INFO:uvicorn.error:Application startup complete.
INFO:watchfiles.main:3 changes detected
INFO:watchfiles.main:1 change detected
WARNING:uvicorn.error:WatchFiles detected changes in 'commands/session.py'. Reloading...
INFO:uvicorn.error:Shutting down
INFO:uvicorn.error:Waiting for application shutdown.
INFO:uvicorn.error:Application shutdown complete.
INFO:uvicorn.error:Finished server process [5575]
INFO:watchfiles.main:1 change detected
INFO:uvicorn.error:Started server process [5943]
INFO:uvicorn.error:Waiting for application startup.
INFO:uvicorn.error:Application startup complete.
INFO:watchfiles.main:3 changes detected
INFO:watchfiles.main:1 change detected
WARNING:uvicorn.error:WatchFiles detected changes in 'commands/session.py'. Reloading...
INFO:uvicorn.error:Shutting down
INFO:uvicorn.error:Waiting for application shutdown.
INFO:uvicorn.error:Application shutdown complete.
INFO:uvicorn.error:Finished server process [5943]
INFO:watchfiles.main:1 change detected
INFO:watchfiles.main:1 change detected
WARNING:uvicorn.error:WatchFiles detected changes in 'commands/session.py'. Reloading...
INFO:watchfiles.main:3 changes detected
INFO:watchfiles.main:1 change detected
WARNING:uvicorn.error:WatchFiles detected changes in 'commands/session.py'. Reloading...
INFO:watchfiles.main:3 changes detected
INFO:watchfiles.main:1 change detected
WARNING:uvicorn.error:WatchFiles detected changes in 'commands/session.py'. Reloading...
INFO:watchfiles.main:4 changes detected
WARNING:uvicorn.error:WatchFiles detected changes in 'commands/session.py'. Reloading...
INFO:watchfiles.main:1 change detected
WARNING:uvicorn.error:WatchFiles detected changes in 'commands/session.py'. Reloading...
INFO:watchfiles.main:3 changes detected
INFO:watchfiles.main:1 change detected
WARNING:uvicorn.error:WatchFiles detected changes in 'commands/session.py'. Reloading...
INFO:watchfiles.main:3 changes detected
INFO:watchfiles.main:1 change detected
WARNING:uvicorn.error:WatchFiles detected changes in 'commands/session.py'. Reloading...
INFO:watchfiles.main:3 changes detected
INFO:watchfiles.main:1 change detected
WARNING:uvicorn.error:WatchFiles detected changes in 'commands/session.py'. Reloading...
INFO:watchfiles.main:3 changes detected
INFO:watchfiles.main:1 change detected
WARNING:uvicorn.error:WatchFiles detected changes in 'commands/session.py'. Reloading...
INFO:watchfiles.main:3 changes detected
INFO:watchfiles.main:1 change detected
WARNING:uvicorn.error:WatchFiles detected changes in 'commands/session.py'. Reloading...
INFO:watchfiles.main:3 changes detected
INFO:watchfiles.main:1 change detected
WARNING:uvicorn.error:WatchFiles detected changes in 'commands/session.py'. Reloading...
INFO:watchfiles.main:3 changes detected
INFO:watchfiles.main:1 change detected
WARNING:uvicorn.error:WatchFiles detected changes in 'commands/session.py'. Reloading...
INFO:watchfiles.main:3 changes detected
INFO:watchfiles.main:1 change detected
WARNING:uvicorn.error:WatchFiles detected changes in 'commands/session.py'. Reloading...
INFO:watchfiles.main:3 changes detected
INFO:watchfiles.main:1 change detected
WARNING:uvicorn.error:WatchFiles detected changes in 'commands/session.py'. Reloading...
INFO:watchfiles.main:3 changes detected
INFO:watchfiles.main:1 change detected
WARNING:uvicorn.error:WatchFiles detected changes in 'commands/command.py'. Reloading...
INFO:watchfiles.main:3 changes detected
INFO:watchfiles.main:1 change detected
WARNING:uvicorn.error:WatchFiles detected changes in 'commands/session.py'. Reloading...
INFO:watchfiles.main:4 changes detected
WARNING:uvicorn.error:WatchFiles detected changes in 'commands/session.py'. Reloading...
INFO:watchfiles.main:3 changes detected
INFO:watchfiles.main:1 change detected
WARNING:uvicorn.error:WatchFiles detected changes in 'commands/command.py'. Reloading...
INFO:watchfiles.main:3 changes detected
INFO:watchfiles.main:1 change detected
WARNING:uvicorn.error:WatchFiles detected changes in 'commands/command.py'. Reloading...
INFO:watchfiles.main:1 change detected
WARNING:uvicorn.error:WatchFiles detected changes in 'commands/command.py'. Reloading...
INFO:watchfiles.main:3 changes detected
INFO:watchfiles.main:2 changes detected
INFO:watchfiles.main:1 change detected
WARNING:uvicorn.error:WatchFiles detected changes in 'commands/save_url.py'. Reloading...
INFO:watchfiles.main:3 changes detected
INFO:watchfiles.main:1 change detected
WARNING:uvicorn.error:WatchFiles detected changes in 'commands/generate_ppt.py'. Reloading...
INFO:uvicorn.error:Stopping reloader process [5495]
